### PR TITLE
File system based indexing 

### PR DESCRIPTION
**Pull request by IljaN**
Early WIP

### CODE REVIEW

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/storage/repo.go  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=6050607&bid=19821115)

File: pkg/storage/repo.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+	var bytes []byte
+	if bytes, err = json.Marshal(a); err != nil {
+		return merrors.InternalServerError(r.serviceID, "could not marshal account: %v", err.Error())
+	}
+
+	path := filepath.Join(r.dataPath, "accounts", a.Id)
+
+	if err = ioutil.WriteFile(path, bytes, 0600); err != nil {
+		return merrors.InternalServerError(r.serviceID, "could not write account: %v", err.Error())
```
Comment (IljaN) [inline pkg/storage/repo.go:42]: Push micro errors out

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/storage/repo.go  4
- pkg/index/manager.go  7
- pkg/index/index.go  4
- pkg/index/manager_test.go  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=6050607&bid=19821115)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/index/manager_test.go  4
- pkg/storage/repo.go  4
- pkg/index/manager.go  9
- pkg/index/primary.go  9
- pkg/index/data_test.go  6
- pkg/index/errors.go  1
- pkg/index/unique_test.go  4
- pkg/index/unique.go  5
         

Clones added
============
- pkg/index/manager_test.go  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=6050607&bid=19821115)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/storage/repo.go  4
- pkg/index/primary.go  9
- pkg/index/manager.go  8
- pkg/index/errors.go  1
- pkg/index/data_test.go  6
- pkg/index/manager_test.go  4
- pkg/index/unique_test.go  4
- pkg/index/reflect.go  4
- pkg/index/unique.go  5
- pkg/index/map.go  2
         

Clones added
============
- pkg/index/manager_test.go  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=6050607&bid=19821115)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/index/map.go  2
- pkg/index/errors.go  1
- pkg/index/manager.go  8
- pkg/index/reflect.go  4
- pkg/index/unique.go  5
- pkg/storage/repo.go  4
- pkg/index/unique_test.go  4
- pkg/index/data_test.go  6
- pkg/index/primary.go  9
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=6050607&bid=19821115)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/index/errors.go  1
- pkg/storage/repo.go  4
- pkg/index/map.go  2
- pkg/index/index_test.go  4
- pkg/index/manager.go  8
- pkg/index/reflect.go  4
- pkg/index/unique.go  5
- pkg/index/primary.go  9
- pkg/index/index.go  5
- pkg/index/unique_test.go  4
- pkg/index/data_test.go  6
         

Clones added
============
- pkg/index/index_test.go  1
- pkg/index/unique.go  2
- pkg/index/index.go  2
- pkg/index/unique_test.go  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=6050607&bid=19821115)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/index/map.go  2
- pkg/index/unique.go  5
- pkg/index/unique_test.go  4
- pkg/index/reflect.go  4
- pkg/index/index.go  5
- pkg/index/data_test.go  6
- pkg/storage/repo.go  4
- pkg/index/errors.go  1
- pkg/index/index_test.go  4
- pkg/index/manager.go  6
         

Clones added
============
- pkg/index/unique.go  2
- pkg/index/unique_test.go  1
- pkg/index/index.go  2
- pkg/index/index_test.go  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=6050607&bid=19821115)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/index/unique.go  5
- pkg/index/data_test.go  6
- pkg/index/reflect.go  4
- pkg/index/map.go  2
- pkg/index/unique_test.go  4
- pkg/index/index_test.go  4
- pkg/index/manager.go  6
- pkg/storage/repo.go  4
- pkg/index/errors.go  1
- pkg/index/index.go  8
         

Clones added
============
- pkg/index/unique.go  2
- pkg/index/unique_test.go  1
- pkg/index/index_test.go  1
- pkg/index/index.go  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=6050607&bid=19821115)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/index/index.go  8
- pkg/index/unique.go  5
- pkg/index/index_test.go  4
- pkg/index/errors.go  1
- pkg/index/map.go  2
- pkg/index/manager.go  6
- pkg/index/reflect.go  4
- pkg/storage/repo.go  4
- pkg/index/manager_test.go  4
- pkg/index/data_test.go  6
- pkg/index/unique_test.go  4
         

Clones added
============
- pkg/index/index.go  2
- pkg/index/unique.go  2
- pkg/index/index_test.go  1
- pkg/index/manager.go  2
- pkg/index/manager_test.go  4
- pkg/index/unique_test.go  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=6050607&bid=19821115)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/index/data_test.go  6
- pkg/storage/repo.go  4
- pkg/index/manager.go  6
- pkg/index/index.go  8
- pkg/index/unique_test.go  4
- pkg/index/map.go  2
- pkg/index/manager_test.go  4
- pkg/index/errors.go  1
- pkg/index/reflect.go  4
- pkg/index/unique.go  5
- pkg/index/index_test.go  4
         

Clones added
============
- pkg/index/manager.go  2
- pkg/index/index.go  2
- pkg/index/unique_test.go  1
- pkg/index/manager_test.go  4
- pkg/index/unique.go  2
- pkg/index/index_test.go  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=6050607&bid=19821115)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/index/index_test.go  4
- pkg/index/errors.go  1
- pkg/index/manager_test.go  4
- pkg/index/unique.go  6
- pkg/index/unique_test.go  4
- pkg/index/map.go  2
- pkg/index/reflect.go  4
- pkg/index/manager.go  6
- pkg/storage/repo.go  4
- pkg/index/data_test.go  6
- pkg/index/index.go  8
         

Clones added
============
- pkg/index/index_test.go  2
- pkg/index/manager_test.go  4
- pkg/index/unique.go  2
- pkg/index/unique_test.go  2
- pkg/index/manager.go  2
- pkg/index/index.go  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=6050607&bid=19821115)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/index/unique_test.go  4
- pkg/index/index.go  8
- pkg/index/index_test.go  4
- pkg/storage/repo.go  4
- pkg/index/unique.go  6
- pkg/index/reflect.go  4
- pkg/index/data_test.go  6
- pkg/index/manager.go  6
- pkg/index/manager_test.go  4
- pkg/index/errors.go  1
- pkg/index/map.go  2
         

Clones added
============
- pkg/index/unique_test.go  2
- pkg/index/index.go  2
- pkg/index/index_test.go  2
- pkg/index/unique.go  2
- pkg/index/manager.go  2
- pkg/index/manager_test.go  4
- pkg/service/v0/service.go  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=6050607&bid=19821115)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/index/unique_test.go  4
- pkg/index/index.go  6
- pkg/index/errors.go  1
- pkg/storage/repo.go  4
- pkg/index/data_test.go  6
- pkg/index/map.go  2
- pkg/index/unique.go  6
- pkg/index/index_test.go  4
- pkg/index/reflect.go  4
- pkg/index/non_unique.go  8
- pkg/index/non_unique_test.go  4
         

Clones added
============
- pkg/index/unique_test.go  2
- pkg/service/v0/service.go  2
- pkg/index/index.go  2
- pkg/index/unique.go  2
- pkg/index/index_test.go  4
- pkg/index/non_unique.go  2
- pkg/index/non_unique_test.go  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=6050607&bid=19821115)

**Comment by IljaN** [comment]
Continued here https://github.com/owncloud/ocis/pull/624

**Pull request closed without merging**